src/neo4j_genai/components/kg_writer.py

```diff
@@ -102,26 +101,27 @@ def _upsert_relationship(self, rel: Neo4jRelationship) -> None:
         """
         # Create the initial relationship
         properties = (
-            "{" + ", ".join(f"{p.key}: {p.value}" for p in rel.properties) + "}"
+            "{" + ", ".join(f"{key}: ${key}" for key in rel.properties.keys()) + "}"
             if rel.properties
             else "{}"
         )
+        parameters = rel.properties
         query = UPSERT_RELATIONSHIP_QUERY.format(
             start_node_id=rel.start_node_id,
```

Comment:
Start and end node IDs could be parameters too, but I don't know if there is any risk in doing it that way.

Reply:
In this context the parameters are just the properties saved to the relationship. Why would we need to include the start and end node IDs here also?

Reply:
They're used in the query later on (if I'm not mistaken): 
```
MATCH (start {{ id: '{start_node_id}' }}), (end {{ id: '{end_node_id}' }})
```
which could be written:
```
MATCH (start {{ id: $start_node_id }}), (end {{ id: $end_node_id }})
```